h/views/account_signup.py

```diff
@@ -53,24 +53,24 @@ def post(self):
             return {"form": self.form.render()}
 
         signup_service = self.request.find_service(name="user_signup")
-        signup_service.signup(
-            username=appstruct["username"],
-            email=appstruct["email"],
-            password=appstruct["password"],
-            privacy_accepted=datetime.datetime.utcnow(),
-        )
-
-        self.request.session.flash(
-            jinja2.Markup(
-                _(
-                    "Please check your email and open the link to activate your "
-                    "account."
-                )
-            ),
-            "success",
-        )
-
-        return httpexceptions.HTTPFound(location=self.request.route_url("index"))
+        try:
+            signup_service.signup(
+                username=appstruct["username"],
+                email=appstruct["email"],
+                password=appstruct["password"],
+                privacy_accepted=datetime.datetime.utcnow(),
+            )
+            heading = "Account registration successful"
+            message = (
+                "Please check your email and open the link to activate your account."
+            )
+        except ConflictError as e:
+            heading = "Account already registered"
+            message = (
+                "{failure_reason} Please check your email and open the link to activate your "
+                "account.".format(failure_reason=e.args[0])
+            )
+        return {"heading": _(heading), "message": _(message)}
```

Comment:
The argument to `h.i18n.TranslationString` (aliased to `_` here) should be the string literal to be translated or a string key for the message, not a variable. The reason is that translation tools scan the source code to identify translatable strings (marked by `_`) and then at runtime  the first argument to `_` is used as a key to lookup the translation to return. See https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/i18n.html